### DISCUSSION TITLE
Power-constrained limits

### DISCUSSION THREAD

**Discussion by mpoehlmann** [scikit-hep/pyhf#1922] [Q&A]
The examples and documentation for producing upper limits with the CLs method are excellent. For comparison, I'm trying to compute upper limits with the power-constrained limit method (https://arxiv.org/abs/1105.3166). Is this possible with the pyhf package?

**Comment by kratsg** [comment]
Interesting (this is new to me). I had a quick read and I will ask @cranmer (one of the authors) to chime in as well and correct me as I have probably misunderstood but... my gut feeling is that you will need to create you own calculator, but you can use `pyhf` to provide the pdf you need, and calculate the PCL yourself with a manual calculator/limit finding routine. Specifically, if you follow through the `pyhf.infer.mle.hypotest` function (https://github.com/scikit-hep/pyhf/blob/8e242c9c829963c714f9b10acd28494f004a2452/src/pyhf/infer/__init__.py#L20) and make something similar to an `AsymptoticCalculator`, you should be able to use `pyhf` partially for this.

We don't provide PCL out of the box, but if you manage to provide a usable calculator that fits into the code, we could possibly be willing to support this as well but I would need to discuss with the other core developers before we can make a statement about the inclusion of such a feature in the future (but I know we would be willing to discuss it of course!).
